hrm-service/src/case/case.ts

```diff
@@ -320,6 +322,16 @@ export const searchCases = async (
     caseFilters.excludedStatuses.push('closed');
   }
   caseFilters.includeOrphans = caseFilters.includeOrphans ?? closedCases ?? true;
+
+  /**
+   * VIEW_CASE permission:
+   * If the user can only view his/her own cases, override caseFilters.counsellors.
+   * The search query already filters the cases given an array of counsellors.
+   */
+  if (canOnlyViewOwnCases) {
```

Comment:
I'll make the following change to the front end:
-  Prevent the counselor from filtering `counselors` on all search forms. Any counselors' select fields will be hidden in the scenario where the user can only view its own cases/contacts.

On the backend, one thing I could do is:
- If `caseFilters.counsellors` is empty/null, make `caseFilters.counsellors = [counselorSid]`
- If `caseFilters.counsellors.includes(counselor)`, make `caseFilters.counsellors = [counselorSid]`
- If `!caseFilters.counsellors.includes(counselor)`, then the search will return zero cases

Again, these would only be possible by hacking Aselo, since the front-end will prevent these scenarios.

Reply:
Yea I know it's a minor thing, just that we don't know how else we might use these APIs going forward, so good to make sure the behaviour follows expectations even in those cases we don't currently expect the UI to throw at it

What you suggest above sounds good - the only difference to the current code is if the search specifies counsellors but not OUR counsellor, return empty result